elastic/kibana
# [Cases] Edit custom fields in the case view page

This PR adds the ability to edit custom fields in the case view page. It also refactors the configuration APIs to use React Query.

## Testing

1. Verify that the configuration in the settings page works as before including starting with an empty deployment
2. Verify that you can edit your custom fields in a case with custom fields
3. Verify that you can set your custom fields in a case without custom fields

<details>
<summary>Add custom field to a case</summary>
```
PATCH /api/cases
{
    "cases": [
        {
            "customFields": [
                {
                    "key": "my_custom_field_a",
                    "type": "text",
                    "field": {
                        "value": [
                            "My custom value A"
                        ]
                    }
                },
                {
                    "key": "my_custom_field_b",
                    "type": "toggle",
                    "field": {
                        "value": [
                            true
                        ]
                    }
                }
            ],
            "id": "ee3374f0-56f5-11ee-9bd9-c901fa345098",
            "version": "WzE4LDFd"
        }
    ]
}
```
</details>


https://github.com/elastic/kibana/assets/7871006/deb626c1-09a2-4761-9eb6-2e376d6ceba8



Connected to https://github.com/elastic/kibana/issues/160236

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
